Rediffed by @akien-mga to match oidn 1.1.0 source.

From 1e42e6db81e1a5270ecc0191c5385ce7e7d978e9 Mon Sep 17 00:00:00 2001
From: Jeremy Wong <jmw@netvigator.com>
Date: Wed, 11 Sep 2019 04:46:53 +0800
Subject: [PATCH] src: initialize members in some structures to prevent compile
 errors with VS2017

addresses "error C3615: constexpr function '...' cannot result in a constant expression" with VS2017
---
 src/cpu/rnn/rnn_reorders.hpp | 2 +-
 src/cpu/simple_concat.hpp    | 6 +++---
 src/cpu/simple_sum.hpp       | 2 +-
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/thirdparty/oidn/mkl-dnn/src/cpu/rnn/rnn_reorders.hpp b/thirdparty/oidn/mkl-dnn/src/cpu/rnn/rnn_reorders.hpp
index 597c63e3f8..ae1551390a 100644
--- a/thirdparty/oidn/mkl-dnn/src/cpu/rnn/rnn_reorders.hpp
+++ b/thirdparty/oidn/mkl-dnn/src/cpu/rnn/rnn_reorders.hpp
@@ -131,7 +131,7 @@ struct rnn_weights_reorder_t : public cpu_primitive_t {
             return status::success;
         }

-        format_tag_t itag_;
+        format_tag_t itag_ = mkldnn_format_tag_undef;

     private:
         void init_scratchpad() {
diff --git a/thirdparty/oidn/mkl-dnn/src/cpu/simple_concat.hpp b/thirdparty/oidn/mkl-dnn/src/cpu/simple_concat.hpp
index 5177275452..057cc3c4c7 100644
--- a/thirdparty/oidn/mkl-dnn/src/cpu/simple_concat.hpp
+++ b/thirdparty/oidn/mkl-dnn/src/cpu/simple_concat.hpp
@@ -96,9 +96,9 @@ struct simple_concat_t: public cpu_primitive_t {
             return status::success;
         }

-        int perm_[MKLDNN_MAX_NDIMS];
-        int iperm_[MKLDNN_MAX_NDIMS];
-        dims_t blocks_;
+        int perm_[MKLDNN_MAX_NDIMS] {};
+        int iperm_[MKLDNN_MAX_NDIMS] {};
+        dims_t blocks_ {};

         dim_t nelems_to_concat(const memory_desc_wrapper &data_d) const {
             const int ndims = data_d.ndims();
